content-service/src/main/java/io/meeds/news/rest/NewsRestResourcesV1.java

```diff
@@ -691,18 +691,22 @@ public Response getNewsByActivityId(@Parameter(description = "Activity id", requ
       @ApiResponse(responseCode = "401", description = "User not authorized to schedule the news"),
       @ApiResponse(responseCode = "500", description = "Internal server error") })
   public Response scheduleNews(@Context
-  HttpServletRequest request, @RequestBody(description = "News object to be scheduled", required = true)
-  News scheduledNews) {
+                               HttpServletRequest request,
+                               @Parameter(description = "News object type to be fetched", required = false)
+                               @QueryParam("type")
+                               String newsObjectType,
+                               @RequestBody(description = "News object to be scheduled", required = true)
+                               News scheduledNews) {
     if (scheduledNews == null || StringUtils.isEmpty(scheduledNews.getId())) {
       return Response.status(Response.Status.BAD_REQUEST).build();
     }
     org.exoplatform.services.security.Identity currentIdentity = ConversationState.getCurrent().getIdentity();
     try {
-      News news = newsService.getNewsById(scheduledNews.getId(), currentIdentity, false);
+      News news = newsService.getNewsById(scheduledNews.getId(), currentIdentity, false, newsObjectType);
```

Comment:
do you need to pass newsObjectType as parameter ?

Reply:
Yes, it needs,  because initially the news object is draft, and then it will become an article!